org.eclipse.scanning.points/src/org/eclipse/scanning/jython/JythonInterpreterManager.java

```diff
@@ -0,0 +1,240 @@
+ * which either is sympathic and working with the GDA configuration or
+ * creates an empty and working interpreter.
+ * 
+ * @author Matthew Gerring
+ *
+ */
+class JythonInterpreterManager {
+
+	private static final Logger logger = LoggerFactory.getLogger(JythonInterpreterManager.class);
+	
+	
+	private static volatile PySystemState configuredState;
+
+	/**
+	 * Call to ensure that an interpreter is set up and configured and
+	 * able to load the relevant bundles.
+	 */
+	public static synchronized void setupSystemState(String... bundleNames) {
+		
+		ClassLoader loader=null;
+		if (configuredState==null) { // Relies on setupSystemState() being called early in the server startup.
+			loader = createJythonClassLoader(PySystemState.class.getClassLoader());
+	 		initializePythonPath(loader); 
+		}
+		
+    	PySystemState state = Py.getSystemState();
+    	if (state==configuredState) return;
+    	
+    	if (loader==null) {
+    		// Then someone else has changed the PySystemState
+    		// They will not have added our 
+			loader = createJythonClassLoader(state.getClassLoader());   // Don't clobber their working.		
+    	}
+	   	state.setClassLoader(loader);
+    	
+     	setJythonPaths(state);       // Tries to ensure that enough of jython is on the path
+     	setSpgGeneratorPaths(state, bundleNames); // Adds the scripts directory from points
+	   	Py.setSystemState(state);
+ 
+	   	configuredState = state;
+	}
+
+	private static void setSpgGeneratorPaths(PySystemState state, String... bundleNames) {
+		
+        File location = getBundleLocation("org.eclipse.scanning.points");
+        state.path.add(new PyString(location.getAbsolutePath() + "/scripts/"));
+        
+        if (bundleNames!=null) {
+        	for (String bundle : bundleNames) {
+                location = getBundleLocation(bundle);
+                state.path.add(new PyString(location.getAbsolutePath() + "/scripts/"));
+			}
+        }
+	}
+
+
+	private static void initializePythonPath(ClassLoader loader) {
+		try {
+	    	String jythonBundleName = System.getProperty("org.eclipse.scanning.jython.osgi.bundle.name", "uk.ac.diamond.jython");
+	        File loc = getBundleLocation(jythonBundleName); // TODO Name the jython OSGi bundle without Diamond in it!
+	        if (loc ==null) return;
+	        File jythonDir = find(loc, "jython");
+	        if (jythonDir ==null) return;
+	        
+	        Properties props = new Properties();
+	    	props.setProperty("python.home", jythonDir.getAbsolutePath());
+	    	props.setProperty("python.console.encoding", "UTF-8"); // Used to prevent: console: Failed to install '': java.nio.charset.UnsupportedCharsetException: cp0.
+	    	props.setProperty("python.options.showJavaExceptions", "true");
+	    	props.setProperty("python.verbose", "warning");
+
+	    	Properties preprops = System.getProperties();
+	    	
+	    	PySystemState.initialize(preprops, props, new String[]{}, loader);
+	    
+	    } catch (Throwable ne) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1181)